libsplinter/src/admin/service/shared.rs

```diff
@@ -1824,6 +2113,81 @@ impl AdminServiceShared {
         Ok(())
     }
 
+    fn validate_disband_circuit(
+        &self,
+        circuit: &Circuit,
+        signer_public_key: &[u8],
+        requester_node_id: &str,
+    ) -> Result<(), AdminSharedError> {
+        if requester_node_id.is_empty() {
+            return Err(AdminSharedError::ValidationFailed(
+                "requester_node_id is empty".to_string(),
+            ));
+        }
+
+        self.validate_key(signer_public_key)?;
+
+        if !self
+            .key_verifier
+            .is_permitted(requester_node_id, signer_public_key)?
+        {
+            return Err(AdminSharedError::ValidationFailed(format!(
+                "{} is not registered for the requester node {}",
+                to_hex(signer_public_key),
+                requester_node_id,
+            )));
+        }
+
+        self.key_permission_manager
+            .is_permitted(signer_public_key, PROPOSER_ROLE)
+            .map_err(|_| {
+                AdminSharedError::ValidationFailed(format!(
+                    "{} is not permitted to vote for node {}",
+                    to_hex(signer_public_key),
+                    requester_node_id
+                ))
+            })?;
+
+        if self.has_proposal(circuit.get_circuit_id())? {
+            return Err(AdminSharedError::ValidationFailed(format!(
+                "Ignoring duplicate disband proposal of circuit {}",
```

Comment:
This should be updated to say "Ignoring duplicate proposal for circuit" (and in the create validation) since it might be a different proposal type